spec/validation/configure_spec.rb

```diff
@@ -34,244 +41,264 @@
     it 'fails with invalid top level keys' do
       h = correct_hash.deep_merge(invalid_key: true)
-      expect(run_configure_validation(h).keys).to eq([:valid_top_level_keys])
+      expect_configure_error(h, /invalid top level key/)
     end
 
-    it 'fails if question is not a hash' do
-      h = correct_hash.deep_merge(group: {
-                                    question: 'Am I missing my mid level question key?',
-                                    type: 'string',
-                                    default: 'Each field will now be interpreted as a separate question',
-                                  })
-      results = build_validator(h).validate.errors
-      expect(results.keys).to eq([:parameters])
-      expect(results[:parameters][0]).to eq('must be a hash')
+    it 'fails if sections are not an array' do
+      h = correct_hash.deep_merge(group: { key: 'I am not an array' })
+      expect_configure_error(h, /must be an array/)
     end
   end
 
   context 'with invalid question fields' do
-    it 'fails if unrecognized fields in a question' do
-      h = correct_hash.deep_merge(domain: {
-                                    invalid_field_question: {
-                                      question: 'Do I have have an unrecognized field?',
-                                      default: 'I do',
-                                      invalid_filed: true,
-                                    },
-                                  })
-      expect(run_configure_validation(h).keys).to eq([:valid_top_level_question_keys])
-    end
-
-    it 'fails if question is missing a title' do
-      h = correct_hash.deep_merge(domain: {
-                                    missing_title_question: {
-                                      default: 'I am missing my question!',
-                                    },
-                                  })
-      results = run_configure_validation(h)
-      expect(results[:parameters][:question][0]).to eq('is missing')
+    it 'fails if the question is missing or empty identifier' do
+      h = correct_hash.deep_merge(self: [{ question: 'I have no identifier' }])
+      expect_configure_error(h, /is missing/)
+      h = correct_hash.deep_merge(self: [{
+                                    question: 'I have no identifier',
+                                    identifier: '',
+                                  }])
+      expect_configure_error(h, /must be filled/)
     end
 
-    it 'fails if question if the title is empty' do
-      h = correct_hash.deep_merge(node: {
-                                    missing_title_question: {
-                                      question: '',
-                                      default: 'I am missing my question!',
-                                    },
-                                  })
-      results = run_configure_validation(h)
-      expect(results[:parameters][:question][0]).to eq('must be filled')
+    it 'fails if question is missing or empty' do
```

Comment:
As above, can we split the two assertions out into two tests?

Reply:
Fixed in: https://github.com/alces-software/metalware/pull/185/commits/571722574003bcda5991dde8768e97f85b9452fa